src/deluge/dsp/delay/delay.cpp

```diff
@@ -17,190 +17,164 @@
 
 #include "dsp/delay/delay.h"
 #include "definitions_cxx.hpp"
+#include "dsp/delay/delay_buffer.h"
 #include "io/debug/log.h"
 #include <cstdlib>
 
-Delay::Delay() {
-	pingPong = true;
-	analog = false;
-	repeatsUntilAbandon = 0;
-	prevFeedback = 0;
-
-	syncLevel = (SyncLevel)5;
-
-	syncType = SYNC_TYPE_EVEN;
-}
-
-void Delay::cloneFrom(Delay* other) {
-	pingPong = other->pingPong;
-	analog = other->analog;
-	syncLevel = other->syncLevel;
-}
-
 void Delay::informWhetherActive(bool newActive, int32_t userDelayRate) {
 
 	bool previouslyActive = isActive();
 
 	if (previouslyActive != newActive) {
-		if (newActive) {
-setupSecondaryBuffer:
-			Error result = secondaryBuffer.init(userDelayRate);
-			if (result != Error::NONE) {
-				return;
-			}
-			prepareToBeginWriting();
-			postLPFL = 0;
-			postLPFR = 0;
-		}
-		else {
+		if (!newActive) {
 			discardBuffers();
+			return;
 		}
+
+setupSecondaryBuffer:
+		Error result = secondaryBuffer.init(userDelayRate);
+		if (result != Error::NONE) {
+			return;
+		}
+		prepareToBeginWriting();
+		postLPFL = 0;
+		postLPFR = 0;
+		return;
 	}
 
 	// Or if no change...
-	else {
 
-		// If it already was active...
-		if (previouslyActive) {
-			// If no writing has happened yet to this Delay, check that the buffer is the right size.
-			// The delay time might have changed since it was set up, and we could be better off making a new buffer,
-			// which is easily done now, before anything's been written
-			if (!primaryBuffer.isActive() && secondaryBuffer.isActive()
-			    && sizeLeftUntilBufferSwap == getAmountToWriteBeforeReadingBegins()) {
+	// If it already was active...
+	if (previouslyActive) {
+		// If no writing has happened yet to this Delay, check that the buffer is the right size.
+		// The delay time might have changed since it was set up, and we could be better off making a new buffer,
+		// which is easily done now, before anything's been written
+		if (!primaryBuffer.isActive() && secondaryBuffer.isActive()
+		    && sizeLeftUntilBufferSwap == getAmountToWriteBeforeReadingBegins()) {
 
-				int32_t idealBufferSize = secondaryBuffer.getIdealBufferSizeFromRate(userDelayRate);
-				idealBufferSize = std::min(idealBufferSize, (int32_t)DELAY_BUFFER_MAX_SIZE);
-				idealBufferSize = std::max(idealBufferSize, (int32_t)DELAY_BUFFER_MIN_SIZE);
+			int32_t idealBufferSize = DelayBuffer::getIdealBufferSizeFromRate(userDelayRate);
+			idealBufferSize = std::min(idealBufferSize, (int32_t)DelayBuffer::kMaxSize);
+			idealBufferSize = std::max(idealBufferSize, (int32_t)DelayBuffer::kMinSize);
```

Comment:
Is it ever correct to use the `IdealBufferSizeFromRate` without this clamping? If not we should fold it in.

Reply:
Done.